### PR TITLE
Show warning dialog when deleting request with donation forms

### PR DESCRIPTION
**Pull request by kevin-wangg**
Deleting a unfulfilled request that is matched with a donationForm that contributes >0 quantity will result in a warning.

https://www.notion.so/uwblueprintexecs/Show-warning-dialog-when-deleting-request-with-donation-forms-8c13581056394dcfb1936948dba79b3c

### CODE REVIEW

File: client/src/components/molecules/RequestsTable.tsx [hunk 1]
```diff
@@ -78,6 +80,28 @@ const RequestsTable: FunctionComponent<Props> = (props: Props) => {
...
     });
+    const handleDeleteRequest = (index: number) => {
+        const req = requests[index];
+        if (req.fulfilledAt != null) {
+            // If Request is fulfilled then don't show warning
+            onDeleteRequest(index);
+        } else {
+            let canDelete = true;
+            const matchedDonations = req.matchedDonations;
+            if (matchedDonations != null) {
```
Comment (Anand-Issac) [inline client/src/components/molecules/RequestsTable.tsx]: I would use strict equality always !==

File: client/src/components/molecules/RequestsTable.tsx [hunk 2]
```diff
@@ -78,6 +80,28 @@ const RequestsTable: FunctionComponent<Props> = (props: Props) => {
             window.location.reload();
         }
     });
+    const handleDeleteRequest = (index: number) => {
+        const req = requests[index];
+        if (req.fulfilledAt != null) {
+            // If Request is fulfilled then don't show warning
```
Comment (Anand-Issac) [inline client/src/components/molecules/RequestsTable.tsx]: nit: comment not needed

**Review by Anand-Issac — APPROVED** [review]
lgtm

File: client/src/components/molecules/RequestsTable.tsx (hunk 2 above)
Comment (kevin-wangg) [reply client/src/components/molecules/RequestsTable.tsx]: Removed.

File: client/src/components/molecules/RequestsTable.tsx (hunk 1 above)
Comment (kevin-wangg) [reply client/src/components/molecules/RequestsTable.tsx]: Updated the logic checking to check if the value is falsely, which includes null and undefined.

**Pull request merged**